docs/sdks/server-sdks/java.md

```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
+  <version>3.0.1</version>
 </dependency>
 ```
 
-### Define an assignment logger
+### Initialize the SDK
 
-If you're using Gradle instead, add it to your `build.gradle` file:
+The SDK is initialized using the builder pattern. 
 
-```groovy
-implementation 'cloud.eppo:eppo-server-sdk:2.4.5'
+Initialize the SDK with an SDK key, which can be generated within the [Eppo interface](https://eppo.cloud/feature-flags/keys):
+
+```java
+EppoClient.Builder()
+  .apiKey(apiKey)
+  .buildAndInit();
 ```
 
-## 2. Initialize the SDK
+Initialization should happen when your application starts up, and generates a singleton client instance to be used 
+throughout the application lifecycle. After initialization, you can access the client with `EppoClient.getInstance()`.
 
-Eppo encourages centralizing application logging as much as possible. Accordingly, instead of implementing a new logging framework, Eppo's SDK integrates with your existing logging system via a logging callback function defined at SDK initialization. This logger takes an [analytic event](/sdks/server-sdks/java/#assignment-logger-schema) created by Eppo, `assignment`, and writes in to a table in the data warehouse (Snowflake, Databricks, BigQuery, or Redshift).
+### Assign variations
 
-The code below illustrates an example implementation of a logging callback using Segment. You could also use your own logging system, the only requirement is that the SDK receives a `logAssignment` function. Here we define an implementation of the Eppo `IAssignmentLogger` interface containing a single function named `logAssignment`:
+After configuring the flag in the [Eppo interface](https://eppo.cloud/feature-flags), you can assign subjects variations.
 
+Assign using `get<Type>Assignment`, with `<Type>` depending on the type of the flag.
+
+These assignment functions take the following parameters:
+- `flagKey` (String): The key of the feature flag corresponding to the bandit
+- `subjectKey` (String): The identifier of the subject (e.g., user) to be assigned a variation
+- `subjectAttributes` (Attributes): _Optional_ - Attributes of the subject, used by targeting rules
+- `defaultValue` (String): The default variation to return if the flag is not successfully evaluated
+
+For example, for a string-valued flag, you would use `getStringAssignment()`:
 
 ```java
-import com.eppo.sdk.dto.IAssignmentLogger;
-import com.eppo.sdk.dto.AssignmentLogData;
-
-public class AssignmentLoggerImpl implements IAssignmentLogger {
-  public void logAssignment(AssignmentLogData event) {
-    analytics.enqueue(TrackMessage.builder("Experiment viewed")
-        .userId(event.subject)
-        .properties(ImmutableMap.builder()
-            .put("experiment", event.experiment)
-            .put("variation", event.variation)
-            .put("timestamp", event.timestamp)
-            .build()
-        )
-    );
-  }
-}
+String assignedVariation = eppoClient.getStringAssignment("subjectKey", "flagkey", "defaultValue");
```

Comment:
Can a non-optional argument follow an optional attribute? How does the compiler know "defaultValue" is not subjectAttributes?

Reply:
That's the beauty of strongly-typed languages like Java. It looks at method _signatures_. So if it sees three `String`s, it knows no attributes are provided. If it sees `String, String, Attributes, String`, it knows attributes are provided. `String, String, Attributes` would not be allowed--default must always be provided.

All that being said, I like loosely typed languages like JavaScript. But this specific case of overloading methods is a perk.